sys/targets/targets.go

```diff
@@ -202,6 +203,21 @@ var List = map[string]map[string]*Target{
 			KernelArch:       "powerpc",
 			KernelHeaderArch: "powerpc",
 		},
+		"s390x": {
+			PtrSize:          8,
+			PageSize:         4 << 10,
+			LittleEndian:     false,
+			Triple:           "s390x-linux-gnu",
+			KernelArch:       "s390",
+			KernelHeaderArch: "s390",
+			SyscallTrampolines: map[string]string{
+				// The s390x Linux syscall ABI allows for upto 5 input parameters passed in registers, and this is not enough
+				// for the mmap syscall. Therefore, all input parameters for the mmap syscall are packed into a struct
+				// on user stack and the pointer to the struct is passed as an input parameter to the syscall.
+				// To work around this problem we therefore reroute the mmap syscall to the glibc mmap wrapper.
+				"mmap": "mmap",
```

Comment:
Good.
I will wait for pkg/csource part. With this map in place, the change in pkg/csource should be something reasonably simple.

Reply:
done

Reply:
Do i need to get rid of __NR_mmap in generated C ?

Reply:
Yes, it can make sense. It can be with sys/targets.Target.NeedSyscallDefine.
The idea behind that is that if we use some very new syscall, the __NR_ const may not be present in host headers yet (say, it was just added to linux-next).